_posts/2021-05-20-tmx1.md

```diff
@@ -145,9 +145,9 @@ In this section we define the methods we will use to analyse replays. To properl
 
 #### 2. Internals of TrackMania input handling
 *Note: if you want to skip technical details, [skip to section 2.3 Ensuring correctness of presented footage](#3-ensuring-correctness-of-presented-footage)*<br><br>
-*Note: the following information contains real symbol names that are used within TrackMania’s code. The knowledge of it’s innerworkings is a result of reverse engineering game’s code.*<br><br>
+*Note: the following information contains real symbol names that are used within TrackMania’s code. The knowledge of its innerworkings is a result of reverse engineering game’s code.*<br><br>
 
-TrackMania Nations and United Forever internally describes it’s subsystems as buffers. Each buffer (internally named `CMwCmdBuffer`) is responsible for running a certain task, such as graphics rendering, audio handling, input handling, physics simulation and more. Each buffer contains a reference to a function it executes (internally named `CMwCmd`). The buffers get managed & executed in a loop by another object called `CMwCmdBufferCore`. The function `CMwCmdBufferCore::Run` gets called repeatedly from the function `CGbxApp::GameLoop`. Whenever in a race, a buffer responsible for handling input gets executed every game tick (via `CMwCmdBuffer::Run`), which is a function called `InputRace`, contained within the `CTrackManiaRace` class. Along with this function, another buffer that calls `CHmsZoneDynamic::PhysicsStep2` is executed to advance race simulation. Both buffers are executed with a frequency of 100/s, which is the simulation tick rate.
+TrackMania Nations and United Forever internally describes its subsystems as buffers. Each buffer (internally named `CMwCmdBuffer`) is responsible for running a certain task, such as graphics rendering, audio handling, input handling, physics simulation and more. Each buffer contains a reference to a function it executes (internally named `CMwCmd`). The buffers get managed & executed in a loop by another object called `CMwCmdBufferCore`. The function `CMwCmdBufferCore::Run` gets called repeatedly from the function `CGbxApp::GameLoop`. Whenever in a race, a buffer responsible for handling input gets executed every game tick (via `CMwCmdBuffer::Run`), which is a function called `InputRace`, contained within the `CTrackManiaRace` class. Along with this function, another buffer that calls `CHmsZoneDynamic::PhysicsStep2` is executed to advance race simulation. Both buffers are executed with a frequency of 100/s, which is the simulation tick rate.
```

Comment:
This doesn't make too much sense either way, buffer is probably meant as event buffer to be consumed in a producer-consumer paradigm (as you then say the buffers are popped in the main thread), but eh whatever